csharp/extractor/Semmle.Util/FileUtils.cs

```diff
@@ -143,5 +143,37 @@ public static string NestPaths(ILogger logger, string? outerpath, string innerpa
             }
             return nested;
         }
+
+        public static string GetTemporaryWorkingDirectory(out bool shouldCleanUp)
```

Comment:
Maybe it makes sense to put these methods in class that implements `IDisposable` to ensure that cleanup happens if needed? Could `TemporaryDirectory` be re-used with some extra exposed methods?

Reply:
The cleanup of the temporary directory is somewhat error prone. At the moment we basically expect the temporary directory to be cleaned up only at (or close to) application exit. Putting it in a more reusable class would suggest that you can dispose it any time, but that would break the application, because we currently store generated source files in this folder too.

Reply:
Fair enough.